interop-testing/src/main/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,341 @@
+import io.grpc.testing.integration.Messages.StreamingInputCallResponse;
+import io.grpc.testing.integration.Messages.StreamingOutputCallRequest;
+import io.grpc.testing.integration.Messages.StreamingOutputCallResponse;
+import io.netty.channel.local.LocalAddress;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalServerChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http2.DefaultHttp2LocalFlowController;
+import io.netty.handler.codec.http2.Http2LocalFlowController;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketOption;
+import java.net.UnknownHostException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.spi.SelectorProvider;
+import java.text.MessageFormat;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+public class NettyFlowControlTest extends AbstractInteropTest {
+
+  private static Logger logger;
+  private static TestLogHandler logHandler;
+
+  private ManagedChannel channel;
+  private InetAddress addr;
+
+  // TODO: make all these arguments
+  private final int port = 5001;
+  private final int proxyPort = 5050;
+  private final String loopback = "127.0.0.1";
+  private final boolean useProxy = true;
+
+  // in kbits
+  private final int LOW_BAND = 1024 * 1024;
+  private final int MED_BAND = 5 * 1024 * 1024;
+  private final int HIGH_BAND = 10 * 1024 * 1024;
+
+  // in milliseconds
+  private final int LOW_LAT = 50;
+  private final int MED_LAT = 200;
+  private final int HIGH_LAT = 500;
+
+  // in bytes
+  private final int TINY_WINDOW = 1;
+  private final int REGULAR_WINDOW = 64 * 1024;
+  private final int BIG_WINDOW = 1024 * 1024;
+
+  private int testsPassed;
+  private int testsFailed;
+
+  private final ThreadPoolExecutor executor =
+      new ThreadPoolExecutor(1, 10, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+  private final TrafficControlProxy proxy =
+      new TrafficControlProxy(LOW_BAND, LOW_LAT, TimeUnit.MILLISECONDS);
+
+
+  public static void main(String[] args) {
```

Comment:
I have a main method because right now I am running these tests from the command line. I was trying to model this based on other integration tests and benchmarks, but should these look more like unit tests?